Cross-Compiling with -d:mingw fails to locate compiler under OSX

<!-- Think about the title, twice. -->

<!-- Summarize the problem here, keep it short and simple. -->

According to the documentation, a command like `nim c  -d:mingw test.nim` should invoke mingw. Unfortunately, it can't find the executable, even if I provide the command 
```
nim c --path:/usr/local/bin -d:mingw test.nim
```
Note: the compiler exists at `/usr/local/bin/x86_64-w64-mingw32-gcc` , not `/usr/bin/x86_64-w64-mingw32-gcc`

### Example
<!-- Paste your example in the code-block below. -->
```nim
echo "Hello World!"
```

### Current Output
```
Hint: used config file '/usr/local/Cellar/nim/0.19.4/nim/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: test [Processing]
CC: test
CC: stdlib_system
Error: execution of an external compiler program '/usr/bin/x86_64-w64-mingw32-gcc -c  -w -mno-ms-bitfields  -I/usr/local/Cellar/nim/0.19.4/nim/lib -o /Users/teras/.cache/nim/test_d/test.c.o /Users/teras/.cache/nim/test_d/test.c' failed with exit code: 127

/bin/sh: /usr/bin/x86_64-w64-mingw32-gcc: No such file or directory

```


### Expected Output
<!-- What should be the correct output? -->

To properly compile!


### Possible Solution
<!--- Have you found a possible solution? Post it here. -->
The compiler should be able to accept the path, and not hard code the path of the copiler?

### Additional Information
<!--- For Example:
* Nim Compiler Version 0.19.4 [MacOSX: amd64]
Compiled at 2019-02-07

* Was it working in the previous Nim releases?
* A link to a related issue or discussion.
-->
```
$ nim -v
Nim Compiler Version 0.19.4 [MacOSX: amd64]
Compiled at 2019-02-07
Copyright (c) 2006-2018 by Andreas Rumpf

active boot switches: -d:release -d:useLinenoise
```

If the config is changed to 
```nim
@if unix and mingw:
  # Cross compile for Windows from Linux/OSX using MinGW
  i386.windows.gcc.exe = "i686-w64-mingw32-gcc"
  i386.windows.gcc.linkerexe = "i686-w64-mingw32-gcc"

  amd64.windows.gcc.exe = "x86_64-w64-mingw32-gcc"
  amd64.windows.gcc.linkerexe = "x86_64-w64-mingw32-gcc"

  @if macosx:
    amd64.windows.gcc.path = "/usr/local/bin"
    i386.windows.gcc.path = "/usr/bin"
  @else:
    i386.windows.gcc.path = "/usr/local/bin"
    amd64.windows.gcc.path = "/usr/bin"
  @end

  os = windows
@end
```

then the system is able to locate the compiler.
But now it fails with this error message:
```
/usr/local/Cellar/mingw-w64/5.0.4_1/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: cannot find -ldl
collect2: error: ld returned 1 exit status
Error: execution of an external program failed: '/usr/local/bin/x86_64-w64-mingw32-gcc   -o /Users/teras/Sources/test.exe  /Users/teras/.cache/nim/test_d/test.c.o /Users/teras/.cache/nim/test_d/stdlib_system.c.o    -ldl'
```

probably due to this part of the config ?
```nim
@if not bsd or haiku:
    # -fopenmp
    gcc.options.linker = "-ldl"
    gcc.cpp.options.linker = "-ldl"
    clang.options.linker = "-ldl"
    clang.cpp.options.linker = "-ldl"
    tcc.options.linker = "-ldl"
  @end
```

If I replace the content of the strings with empty strings, it works, but I can't just comment them out.

--- a/config/nim.cfg
+++ b/config/nim.cfg
@@ -85,15 +85,28 @@ path="$lib/pure"
 
 @if unix and mingw:
   # Cross compile for Windows from Linux/OSX using MinGW
-  os = windows
-
-  i386.windows.gcc.path = "/usr/bin"
   i386.windows.gcc.exe = "i686-w64-mingw32-gcc"
   i386.windows.gcc.linkerexe = "i686-w64-mingw32-gcc"
+  i386.windows.gcc.cpp.exe = "i686-w64-mingw32-g++"
+  i386.windows.gcc.cpp.linkerexe = "i686-w64-mingw32-g++"
 
-  amd64.windows.gcc.path = "/usr/bin"
   amd64.windows.gcc.exe = "x86_64-w64-mingw32-gcc"
   amd64.windows.gcc.linkerexe = "x86_64-w64-mingw32-gcc"
+  amd64.windows.gcc.cpp.exe = "x86_64-w64-mingw32-g++"
+  amd64.windows.gcc.cpp.linkerexe = "x86_64-w64-mingw32-g++"
+
+  @if macosx:
+    i386.windows.gcc.path = "/usr/local/bin"
+    amd64.windows.gcc.path = "/usr/local/bin"
+  @else:
+    i386.windows.gcc.path = "/usr/bin"
+    amd64.windows.gcc.path = "/usr/bin"
+  @end
+
+  os = windows
+
+  gcc.options.linker = ""
+  gcc.cpp.options.linker = ""
 @end
 
 @if unix:

